### PR TITLE
[WIP] e2e framework: remove internal k8s.io/kubernetes depedencies for staging

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: andrewsykim <kim.andrewsy@gmail.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
WIP, do not merge. Still evaluating where it's appropriate to duplicate code as part of https://github.com/kubernetes/kubernetes/issues/74352 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by tanjunchen** [comment]
My understanding is there is a direct reference in these files.
```
util.go:	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
util.go:	"k8s.io/kubernetes/pkg/client/conditions"
util.go:	"k8s.io/kubernetes/pkg/controller"
util.go:	"k8s.io/kubernetes/pkg/master/ports"
util.go:	taintutils "k8s.io/kubernetes/pkg/util/taints"
ssh/ssh.go:	sshutil "k8s.io/kubernetes/pkg/ssh"
metrics/metrics_grabber.go:	api "k8s.io/kubernetes/pkg/apis/core"
metrics/metrics_grabber.go:	"k8s.io/kubernetes/pkg/master/ports"
metrics/kubelet_metrics.go:	dockermetrics "k8s.io/kubernetes/pkg/kubelet/dockershim/metrics"
metrics/kubelet_metrics.go:	kubeletmetrics "k8s.io/kubernetes/pkg/kubelet/metrics"
deployment/fixtures.go:	deploymentutil "k8s.io/kubernetes/pkg/controller/deployment/util"
skipper/skipper.go:	"k8s.io/kubernetes/pkg/features"
pod/resource.go:	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
pod/resource.go:	"k8s.io/kubernetes/pkg/client/conditions"
pod/resource.go:	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
pod/wait.go:	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
pod/wait.go:	"k8s.io/kubernetes/pkg/kubelet/util/format"
autoscaling/autoscaling_utils.go:	api "k8s.io/kubernetes/pkg/apis/core"
node/resource.go:	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
node/resource.go:	nodectlr "k8s.io/kubernetes/pkg/controller/nodelifecycle"
node/resource.go:	schedulernodeinfo "k8s.io/kubernetes/pkg/scheduler/nodeinfo"
statefulset/fixtures.go:	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
statefulset/rest.go:	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
statefulset/wait.go:	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
service/jig.go:	"k8s.io/kubernetes/pkg/registry/core/service/portallocator"
providers/kubemark/kubemark.go:	"k8s.io/kubernetes/pkg/kubemark"
security/apparmor.go:	"k8s.io/kubernetes/pkg/security/apparmor"
kubelet/config.go:	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
kubelet/kubelet_pods.go:	"k8s.io/kubernetes/pkg/master/ports"
kubelet/stats.go:	kubeletstatsv1alpha1 "k8s.io/kubernetes/pkg/kubelet/apis/stats/v1alpha1"
kubelet/stats.go:	dockermetrics "k8s.io/kubernetes/pkg/kubelet/dockershim/metrics"
kubelet/stats.go:	"k8s.io/kubernetes/pkg/master/ports"
pods.go:	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
pods.go:	"k8s.io/kubernetes/pkg/kubelet/events"
pods.go:	"k8s.io/kubernetes/pkg/kubelet/sysctl"
resource/runtimeobj.go:	appsinternal "k8s.io/kubernetes/pkg/apis/apps"
resource/runtimeobj.go:	batchinternal "k8s.io/kubernetes/pkg/apis/batch"
resource/runtimeobj.go:	api "k8s.io/kubernetes/pkg/apis/core"
resource/runtimeobj.go:	extensionsinternal "k8s.io/kubernetes/pkg/apis/extensions"
rc/rc_utils.go:	api "k8s.io/kubernetes/pkg/apis/core"
psp.go:	"k8s.io/kubernetes/pkg/security/podsecuritypolicy/seccomp"
test_context.go:	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
job/wait.go:	jobutil "k8s.io/kubernetes/pkg/controller/job"
resource_usage_gatherer.go:	kubeletstatsv1alpha1 "k8s.io/kubernetes/pkg/kubelet/apis/stats/v1alpha1"
resource_usage_gatherer.go:	"k8s.io/kubernetes/pkg/master/ports"
pv/pv.go:	storageutil "k8s.io/kubernetes/pkg/apis/storage/v1/util"
pv/pv.go:	"k8s.io/kubernetes/pkg/volume/util"
```

File: test/e2e/framework/job/wait.go [hunk 1]
```diff
@@ -65,10 +65,19 @@ func WaitForJobFinish(c clientset.Interface, ns, jobName string) error {
 		if err != nil {
 			return false, err
 		}
-		return jobutil.IsJobFinished(curr), nil
+		return isJobFinished(curr), nil
 	})
 }
 
+func isJobFinished(j *batchv1.Job) bool {
```
Comment (tanjunchen) [inline test/e2e/framework/job/wait.go]: How to deal with the [original function](  https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/utils.go#L24), is it copying this function now?

**Comment by oomichi** [comment]
Thanks for doing this, I prefer this PR.

/cc @oomichi

**Comment by andrewsykim** [comment]
@tanjunchen we only care about test/e2e/framework (for now)

**Comment by andrewsykim** [comment]
@oomichi I was planning to break this up into smaller PRs, would you prefer 1 big one instead?

**Comment by alejandrox1** [comment]
I'd opt for smaller PRs.
We should make sure things don't slip by

**Comment by tanjunchen** [comment]
get it.

**Comment by oomichi** [comment]
> @oomichi I was planning to break this up into smaller PRs, would you prefer 1 big PR instead?

I prefer small PRs instead of huge one to avoid code conflicts :-)

**Pull request closed without merging**